protocol/pubsub/v2/protocol.go

```diff
@@ -233,8 +233,14 @@ var _ protocol.Sender = (*Protocol)(nil)
 var _ protocol.Receiver = (*Protocol)(nil)
 var _ protocol.Closer = (*Protocol)(nil)
 
+type withInitialMessageAttributes struct{}
```

Comment:
Hm, something that worries me is that we're adding a new context attribute, but it's taking effect in a separate part of that protocol.

If you see where/how `withOrderingKey` is used [here](https://github.com/cloudevents/sdk-go/blob/af1a06c50c614e4455e902b5e05d562bfc5fabd1/protocol/pubsub/v2/protocol.go#L115-L120), I'm wondering if there's some opportunity for us to clean up/consolidate **or** move that attributes logic to the file where it's actually used? Having a hard time keeping track of the implementation of this protocol tbh (not to blame you btw!)